libyul/YulStack.cpp

```diff
@@ -71,15 +71,35 @@ CharStream const& YulStack::charStream(std::string const& _sourceName) const
 	return *m_charStream;
 }
 
+bool YulStack::parse(std::string const& _sourceName, std::string const& _source)
+{
+	try
+	{
+		m_charStream = std::make_unique<CharStream>(_source, _sourceName);
+		std::shared_ptr<Scanner> scanner = std::make_shared<Scanner>(*m_charStream);
+		m_parserResult = ObjectParser(m_errorReporter, languageToDialect(m_language, m_evmVersion)).parse(scanner, false);
+	}
+	catch (UnimplementedFeatureError const& _error)
+	{
+		solAssert(_error.comment(), "Unimplemented feature errors must include a message for the user");
+		reportUnimplementedFeatureError(_error);
+	}
+
+	if (m_errorReporter.errors().empty())
```

Comment:
```suggestion
	if (!m_errorReporter.hasErrors())
```

Oh, this is actually a minor bug (not in the PR - it was here already). Checking `errors().empty()` will count warnings and infos as errors, which is wrong - those should not fail compilatiohn. Turns out we don't have any warnings in Yul parser though, which is probably why it went unnoticed, but we should fix it nevertheless.

I see there are overall 16 uses of `errors().empty()` in the codebase and some of them might be intended but some look like bugs. Maybe it would be better to fix them in a separate PR. At least this one had no user-visible effects (and the fix does not require a changelog entry) but others might have.

Reply:
Fixed.

Reply:
This is something you should at least put in a separate commit with its own description. It shouldn't be mixed in with other, unrelated changes, especially in a refactor like this. Like I said, I'd even go as far as to leave it as is here and only fix it in a separate PR along with the other instances.

Reply:
Ah right, I was thinking about checking the other cases later in another PR. I removed this change and will leave it for this future PR.

Reply:
[PR](https://github.com/ethereum/solidity/pull/15203) created.